jvm/sargon-desktop-release/build.gradle.kts

```diff
@@ -23,10 +24,37 @@ dependencies {
 
 publishing {
     publications {
+
+        fun command(command: String): String {
+            val out = ByteArrayOutputStream()
+            exec {
+                commandLine(command.split(" "))
+                standardOutput = out
+            }.assertNormalExitValue()
+            return String(out.toByteArray(), Charsets.UTF_8).trim()
+        }
+
         register<MavenPublication>("release") {
             groupId = "com.radixdlt.sargon"
             artifactId = "sargon-desktop-bins"
-            version = System.getenv("SARGON_VERSION")
+
+            val toml = File(projectDir.parentFile.parentFile, "Cargo.toml").readText()
+            val matchResult: MatchResult? = "version\\s*=\\s*\"(.+)\"".toRegex().find(toml)
```

Comment:
this finds the first `version` right? maybe later you can add a depenedency on a TOML file parser in Kotlin, maybe https://github.com/akuleshov7/ktoml and then do `toml[package][version]` or similar.

Reply:
Let me see if I can squeeze that now

Reply:
You are right. No need for such gimmicks. I wrote a more elegant solution hidden in gradle's `buildSrc`